File: ruby/version1_soap.rb
# http://localhost:4567/?n=10
# ruby version1_soap.rb

require 'sinatra'
require 'savon'

set :port, 4567

get '/' do
  n = params['n']
  halt 400, 'Proporciona un número: http://localhost:4567/?n=10' if n.nil?

  client = Savon.client(wsdl: 'https://www.dataaccess.com/webservicesserver/NumberConversion.wso?WSDL')
  response = client.call(:number_to_words, message: { ubi_num: n.to_i })
  result = response.body[:number_to_words_response][:number_to_words_result]

  content_type 'text/plain; charset=utf-8'
  result
end

File: nodejs/version1_soap.js
// Uso: node version1_soap.js 10
// O con servidor: node version1_soap.js server
// http://localhost:3000/?n=10

const http = require('http');
const soap = require('soap');

const WSDL = 'https://www.dataaccess.com/webservicesserver/NumberConversion.wso?WSDL';

const server = http.createServer(async (req, res) => {
  const url = new URL(req.url, `http://localhost:3000`);
  const n = url.searchParams.get('n');

  if (!n) {
    res.end('Proporciona un número: http://localhost:3000/?n=10');
    return;
  }

  const client = await soap.createClientAsync(WSDL);
  const result = await client.NumberToWordsAsync({ ubiNum: parseInt(n) });
  res.end(result[0].NumberToWordsResult);
});

server.listen(3000, () => console.log('Servidor en http://localhost:3000/?n=10'));